### PR TITLE
server: use an individual package for tso

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve? 
Part of modularizing PD. It can be reviewed after the following PRs are merged.
- [x] #1622
- [x] #1623 

### What is changed and how it works?
This PR uses an individual package for tso.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: pkg/typeutil/time.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+func ParseTimestamp(data []byte) (time.Time, error) {
+	nano, err := BytesToUint64(data)
+	if err != nil {
+		return ZeroTime, err
+	}
+
+	return time.Unix(0, int64(nano)), nil
+}
+
+// SubTimeByWallClock returns the duration between two different time.
```
Comment (disksing) [inline pkg/typeutil/time.go]: two different timestamps

Comment (disksing) [reply pkg/typeutil/time.go]: I think better more info to describe why `time.Sub` does not fit our requirement.

File: pkg/typeutil/convension.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+package typeutil
+
+import (
+	"encoding/binary"
+
+	"github.com/pkg/errors"
+)
+
+// BytesToUint64 is transfer bytes to uint64.
```
Comment (disksing) [inline pkg/typeutil/convension.go]: Seems strange. May need some help from i18n :)

File: pkg/typeutil/convension.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+func BytesToUint64(b []byte) (uint64, error) {
+	if len(b) != 8 {
+		return 0, errors.Errorf("invalid data, must 8 bytes, but %d", len(b))
+	}
+
+	return binary.BigEndian.Uint64(b), nil
+}
+
+// Uint64ToBytes is transfer uint64 to bytes.
```
Comment (disksing) [inline pkg/typeutil/convension.go]: Seems strange. May need some help from i18n :)

File: pkg/typeutil/time.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+// limitations under the License.
+
+package typeutil
+
+import "time"
+
+// ZeroTime is a zero time.
+var ZeroTime = time.Time{}
+
+// ParseTimestamp returns a time for a given byte slice.
```
Comment (disksing) [inline pkg/typeutil/time.go]: a timestamp?

File: pkg/typeutil/convension.go (hunk 2 above)
Comment (shafreeck) [reply pkg/typeutil/convension.go]: BytesToUint64 parses a slice of byte to uint64?

File: pkg/typeutil/time_test.go [hunk 5]
```diff
@@ -0,0 +1,36 @@
+package typeutil
```
Comment (nolouch) [inline pkg/typeutil/time_test.go]: License.

**Pull request merged**